
From: Pavel Machek <pavel@ucw.cz>

This function will break with -mregparm, so mark it asmlinkage.


---

 25-akpm/kernel/power/pmdisk.c |    2 +-
 1 files changed, 1 insertion(+), 1 deletion(-)

diff -puN kernel/power/pmdisk.c~pmdisk-needs-asmlinkage kernel/power/pmdisk.c
--- 25/kernel/power/pmdisk.c~pmdisk-needs-asmlinkage	Mon Apr  5 15:23:29 2004
+++ 25-akpm/kernel/power/pmdisk.c	Mon Apr  5 15:23:29 2004
@@ -35,7 +35,7 @@
 #include "power.h"
 
 
-extern int pmdisk_arch_suspend(int resume);
+extern asmlinkage int pmdisk_arch_suspend(int resume);
 
 #define __ADDRESS(x)  ((unsigned long) phys_to_virt(x))
 #define ADDRESS(x) __ADDRESS((x) << PAGE_SHIFT)

_
